packages/devtools_server/lib/src/handlers.dart

```diff
@@ -277,23 +343,60 @@ class DevToolsUsage {
     return properties['enabled'];
   }
 
-  int get surveyShownCount {
-    if (properties['surveyShownCount'] == null) {
-      properties['surveyShownCount'] = 0;
+  bool surveyNameExists(String surveyName) => properties[surveyName] != null;
+
+  void _addSurvey(String surveyName) {
+    assert(activeSurvey != null);
+    assert(activeSurvey == surveyName);
+    rewriteActiveSurvey(false, 0);
+  }
+
+  String get activeSurvey => _activeSurvey;
+
+  set activeSurvey(String surveyName) {
+    assert(surveyName != null);
+    _activeSurvey = surveyName;
+
+    if (!surveyNameExists(activeSurvey)) {
+      // Create the survey if property is non-existent in ~/.devtools
+      _addSurvey(activeSurvey);
     }
+  }
+
+  /// Need to rewrite the entire survey structure for property to be persisted.
+  void rewriteActiveSurvey(bool actionTaken, int shownCount) {
```

Comment:
comment still stands about making these optional named parameters so that you don't have to specify an unchanged value from the call site.